specs/phase1/validator.md

```diff
@@ -0,0 +1,569 @@
+    signatures = []
+    for block_index, custody_bits in enumerate(attestation.custody_bits_blocks):
+        custody_bit = custody_bits[index_in_committee]
+        signing_root = compute_signing_root(
+            AttestationCustodyBitWrapper(
+                attestation_data_root=attestation_data_root,
+                block_index=block_index,
+                bit=custody_bit,
+            ),
+            domain,
+        )
+        signatures.append(bls.Sign(privkey, signing_root))
+
+    return bls.Aggregate(signatures)
+```
+
+### Attestation Aggregation
+
+Some validators are selected to locally aggregate attestations with a similar `attestation_data` to their constructed `attestation` for the assigned `slot`.
+
+Aggregation selection and the core of this duty are largely unchanged from Phase 0. Any additional components or changes are noted.
+
+#### Broadcast aggregate
+
+Note the timing of when to broadcast aggregates is altered in Phase 1+.
+
+If the validator is selected to aggregate (`is_aggregator`), then they broadcast their best aggregate as a `SignedAggregateAndProof` to the global aggregate channel (`beacon_aggregate_and_proof`) three-fourths of the way through the `slot`-that is, `SECONDS_PER_SLOT * 3 / 4` seconds after the start of `slot`.
+
+##### `AggregateAndProof`
+
+`AggregateAndProof` is unchanged other than the contained `Attestation`.
+
+```python
+class AggregateAndProof(Container):
+    aggregator_index: ValidatorIndex
+    aggregate: Attestation
+    selection_proof: BLSSignature
+```
+
+##### `SignedAggregateAndProof`
+
+`AggregateAndProof` is unchanged other than the contained `AggregateAndProof`.
+
+```python
+class SignedAggregateAndProof(Container):
+    message: AggregateAndProof
+    signature: BLSSignature
+```
+
+### Light client committee
+
+In addition to the core beacon chain responsibilities, Phase 1 adds an additional role -- the Light Client Committee -- to aid in light client functionality.
+
+Validators serve on the light client committee for `LIGHT_CLIENT_COMMITTEE_PERIOD` epochs and the assignment to be on a committee is known `LIGHT_CLIENT_COMMITTEE_PERIOD` epochs in advance.
+
+#### Preparation
+
+When `get_current_epoch(state) % LIGHT_CLIENT_COMMITTEE_PERIOD == LIGHT_CLIENT_COMMITTEE_PERIOD - LIGHT_CLIENT_PREPARATION_EPOCHS` each validator must check if they are in the next period light client committee by calling `is_in_next_light_client_committee()`.
+
+If the validator is in the next light client committee, they must join the `light_client_votes` pubsub topic to begin duties at the start of the next period.
+
+```python
+def is_in_next_light_client_committee(state: BeaconState, index: ValidatorIndex) -> bool:
+    current_source_epoch = compute_committee_source_epoch(get_current_epoch(state), LIGHT_CLIENT_COMMITTEE_PERIOD)
+    next_source_epoch = current_source_epoch + LIGHT_CLIENT_COMMITTEE_PERIOD
+    next_committee = get_light_client_committee(state, next_source_epoch)
```

Comment:
The input `epoch` of `get_light_client_committee` should be the epoch that the light clients have to vote for, not the source epoch.

I suppose it should be:
```suggestion
    next_committee = get_light_client_committee(state,get_current_epoch(state) + LIGHT_CLIENT_COMMITTEE_PERIOD)
```

Reply:
I'd actually argue that the above logic is the same, and I considered writing it as you had originally but wasn't sure if the logic was clear.

Making the change regardless. (curious if I am incorrect in saying that you'll get the same result either way)

Reply:
@djrtwo 
No, they are not the same result. I argued the function logic correctness:
- `get_light_client_committee` itself will compute the correct source epoch internally so we don't have to pass the source epoch to it.
- The `epoch` we pass to `get_light_client_committee` should have been the epoch that the light client is going to vote for. In this case, set it to `get_current_epoch(state) + LIGHT_CLIENT_COMMITTEE_PERIOD` can make it forward to the next period.

Reply:
But getting the `current_source_epoch` and adding `LIGHT_CLIENT_COMMITTEE_PERIOD` to it will put you certainly in the next period. Just as `get_current_epoch + LIGHT_CLIENT_COMMITTEE_PERIOD` will also put you in the next period